zone/raids.cpp

```diff
@@ -147,66 +164,129 @@ void Raid::AddMember(Client *c, uint32 group, bool rleader, bool groupleader, bo
 		}
 	}
 
-	Raid *raid_update = nullptr;
-	raid_update = c->GetRaid();
+	auto* raid_update = c->GetRaid();
 	if (raid_update) {
 		raid_update->SendHPManaEndPacketsTo(c);
 		raid_update->SendHPManaEndPacketsFrom(c);
 	}
 
 	auto pack = new ServerPacket(ServerOP_RaidAdd, sizeof(ServerRaidGeneralAction_Struct));
-	ServerRaidGeneralAction_Struct *rga = (ServerRaidGeneralAction_Struct*)pack->pBuffer;
-	rga->rid = GetID();
-	strn0cpy(rga->playername, c->GetName(), 64);
-	rga->zoneid = zone->GetZoneID();
+	auto* rga = (ServerRaidGeneralAction_Struct*) pack->pBuffer;
+	strn0cpy(rga->playername, c->GetName(), sizeof(rga->playername));
+	rga->rid         = GetID();
+	rga->zoneid      = zone->GetZoneID();
+	rga->instance_id = zone->GetInstanceID();
+	worldserver.SendPacket(pack);
+	safe_delete(pack);
+}
+
+void Raid::AddBot(Bot* b, uint32 group, bool rleader, bool groupleader, bool looter) {
+	if (!b) {
+		return;
+	}
+
+	const auto query = fmt::format(
+		"INSERT INTO raid_members SET raidid = {}, "
+		"charid = 0, botid = {}, groupid = {}, _class = {}, level = {}, name = '{}', "
+		"isgroupleader = {}, israidleader = {}, islooter = {}",
+		GetID(),
+		b->GetBotID(),
+		group,
+		b->GetClass(),
+		b->GetLevel(),
+		b->GetName(),
+		groupleader,
+		rleader,
+		looter
+	);
+
+	auto results = database.QueryDatabase(query);
+	if (!results.Success()) {
+		LogError("Error inserting into raid members: [{}]", results.ErrorMessage().c_str());
```

Comment:
This will emit an error when the query fails, redundant

Reply:
removed 204/205 for next push.